server/plugins/pay.js

```diff
@@ -47,15 +48,17 @@ const pay = {
         method: 'get',
         path: '/status',
         handler: async (request, h) => {
-          const pay = request.yar.get('pay')
+          const { pay } = await Cache.getState(request)
           const basePath = request.yar.get('basePath')
           if (pay) {
             const { self, reference } = pay
             const { state } = await payStatus(self)
             if (state.finished) {
               switch (state.status) {
                 case 'success':
-                  let response = await caseManagementPostRequest(request.yar.get('caseManagementData'))
+                  let { caseManagementData } = await Cache.getState(request)
+                  let response = await caseManagementPostRequest(caseManagementData)
+                  await Cache.clearState(request)
```

Comment:
What happens if API times out (but returns a 202)? Maybe add a card for this and talk to Rich. Scenario is we have their data but it's taking loonger to submit to the case management system than we expect. There's no need for the user to resubmit.

Reply:
it should be ok since it's 2xx, hapi/wreck will throw 4/5xx errors

Reply:
i mean, you won't get a reference back, the response body will be empty, so we need to redirect to a different page with appropriate messaging (you can leave this for another task/PR)